app/models/user.rb

```diff
@@ -11,6 +11,19 @@ class User < ApplicationRecord
   validates :email, uniqueness: true
   validates_presence_of :first_name, :last_name
 
+  def self.create_with_omniauth(auth)
+
+    user = find_or_create_by(uid: auth['uid'], provider:  auth['provider'])
+    user.email = "#{auth['uid']}@#{auth['provider']}.com"
+    user.name = auth['info']['name']
+    if User.exists?(user)
+      user
+    else
+      user.save!
```

Comment:
think we can just do a `user.save!` and return, the if/else is superfluous. If the data hasn't changed and you save, it doesn't hurt. You're doing a round trip to check on User.exists anyway, so it's not any less efficient.